# Игра: Угадай число
#
# Используя модуль `random`, напишите программу, которая случайным образом выбирает число от 0 до 100.
# У пользователя есть 6 попыток, чтобы угадать это число. Если пользователь угадывает число,
# выводится сообщение об успехе: "Отлично! Вы угадали число … с … попытки!". Если не угадывает за 6 попыток,
# выводится сообщение о неудаче: “К сожалению, вы не угадали число. Загаданное число было: …”.
#
# В конце программы должны выводиться все попытки пользователя и загаданное число.
#
# По ходу игры после каждой попытки пользователя компьютер выводит сообщение, было ли число пользователя
# больше или меньше загаданного числа: "Загаданное число больше.", "Загаданное число меньше." соответственно.
import random
import datetime

def log_action(message):
    with open("guess_number_log.txt", "a", encoding="utf-8") as log_file:
        log_file.write(f"{datetime.datetime.now()}: {message}\n")

# FINAL
def play_guess_number():
    print()
    print('Добро пожаловать в игру "Угадай число !')
    print('Я загадал число от 0 до 100 , попробуйте угадать его за 6 попыток . Удачи ! ')
    print()
    rand = random.randint(1,100)

    log_action("Начало игры")
    attempts = 0
    lis = []
    for attempts in range(1, 7):
        try:
            guess = int(input('Введите свое предполагаемое число :'))
            lis.append(guess)
            if len(lis) > 6:
                print("Вы превысили количество попыток ! ")
                log_action(f"Игрок не угадал число {guess} с {attempts} попыток.")
                break

            log_action(f"Попытка {attempts}: Введено число {guess}")

            if guess > rand:
                print('К сожалению , Вы не угадали . \n Загаданое число меньше . \n Попробуйте еще раз )')
                log_action(f"Игрок не угадал число. Загаданное число было меньше.")
            elif guess < rand:
                print('К сожалению , Вы не угадали . \n Загаданое число больше . \n Попробуйте еще раз )')
                log_action(f"Игрок не угадал число. Загаданное число было больше.")
            else:
                print(f"Отлично! Вы угадали число с {len(lis)} попытки!")
                print()
                print()
                log_action(f"Игрок угадал число {guess} с {attempts} попытки.")

                input("Игра закончена. Клавиша ввод - дальше в меню")
                from main import main
                main()
        except ValueError as e:
            print(e)
    print("Вы не угадали число ")
    print(f"Ваши попытки : {lis} ")
    print (f"Число , которое я загадал : {rand}")
    log_action(f"Игрок не угадал число с {attempts} попыток. Число , которое я загадал : {rand}")